Web api contract first as abstract controllers

gozdebozkurt opened the pull request:
WrapResponses bugfix - using System.Net.Http carried on FileTemplate
New futures added GenerateControllerInterfaces and GenerateAbstractController in SwaggerToCSharpControllerCommand and SwaggerToCSharpController

Commit:
WrapResponses bugfix - using System.Net.Http carried on FileTemplate from ControllerTemplate

Commit:
SwaggerToWebApiController new futures- GenerateControllerInterfaces and GenerateAbstractController setting is added + created new test classes

Commit:
SwaggerToCSharpControllerCommand new futures- GenerateControllerInterfaces and GenerateAbstractController setting is added

GiampaLab commented on src/NSwag.CodeGeneration.CSharp.Tests/GenerateAbstractControllersTests.cs:
```diff
@@ -0,0 +1,70 @@
+            var code = codeGen.GenerateFile();
+
+            //// Assert
+            Assert.IsTrue(code.Contains("abstract class TestController"));
```
Shouldn't this be `Assert.IsFalse(code.Contains("abstract class TestController"));`

GiampaLab commented on src/NSwag.CodeGeneration.CSharp.Tests/GenerateAbstractControllersTests.cs:
```diff
@@ -0,0 +1,70 @@
+            //// Act
+            var codeGen = new SwaggerToCSharpClientGenerator(document, new SwaggerToCSharpClientGeneratorSettings
+            {
+                GenerateControllerInterfaces = true,
```
Since you are testing only the abstract controller setting here I would keep `GenerateControllerInterfaces = false`

GiampaLab commented on src/NSwag.CodeGeneration.CSharp.Tests/GenerateAbstractControllersTests.cs:
```diff
@@ -0,0 +1,70 @@
+
+            //// Assert
+            Assert.IsTrue(code.Contains("abstract class TestController"));
+            Assert.IsTrue(code.Contains("abstract Foo"));
```
This one also should be false

GiampaLab commented on src/NSwag.CodeGeneration.CSharp/Templates/ControllerTemplate.tt:
```diff
@@ -15,47 +13,62 @@ public interface I<#=Model.Class#>Controller
 <#          if (Model.WrapResponses){#>
-        var result = await _implementation.<#=operation.ActualOperationName#>Async(<#foreach(var parameter in operation.Parameters){#><#=parameter.VariableName#><#if(!parameter.IsLast){#>, <#}}#>);
+        var result = await _implementation.<#=operation.ActualOperationName#>Async(<#foreach(var parameter in operation.Parameters){#><#=parameter.VariableName#><# if(!parameter.IsLast){#>, <#}}#>);
 
```
You can probably revert formatting changes on this line and on lines 48 and 55 as well

Commit:
Code review - fixed test errors
https://github.com/Aqovia/NSwag/pull/1/files/e7186e06b90bc358140c43d5ed9867104da7c8ea#r132965358
https://github.com/Aqovia/NSwag/pull/1/files/e7186e06b90bc358140c43d5ed9867104da7c8ea#r132965807

Commit:
Code review - fixed alligment
https://github.com/Aqovia/NSwag/pull/1/files/e7186e06b90bc358140c43d5ed9867104da7c8ea#r132967498

Commit:
Code review-  unit tests fixed
https://github.com/Aqovia/NSwag/pull/1/files#r132965358

Commit:
Bug fix - WrapResponses using System.Net.Http

Commit:
GenerateControllerInterfaces and GenerateAbstractController settings are deleted. ControllerGenerationFormat parameter is added. (default:partial)Test classes are deleted and new test class is added.

ma499 reviewed (approved):
Looks good to me.
